plugins/scaffolder/src/api.ts

```diff
@@ -136,6 +151,24 @@ export class ScaffolderClient implements ScaffolderApi {
     return schema;
   }
 
+  registerCustomField(field: CustomField) {
+    this.customFields.set(field.name, field);
```

Comment:
Would say we want validation that there are no duplicates here

Reply:
I guess one thing to note however, is that having this validation in place would cause more effort to upgrade as users will have to register the `OwnerPicker` and `RepoUrlPicker` themselves. But I think that's fine and less magic, otherwise people wouldn't be able to override them for their own use cases.